pkg/eval/glob.go

```diff
@@ -208,15 +246,24 @@ func doGlob(gp GlobPattern, abort <-chan struct{}) ([]interface{}, error) {
 	}
 
 	vs := make([]interface{}, 0)
-	if !gp.Glob(func(name string) bool {
+	if !gp.Glob(func(pathInfo glob.PathInfo) bool {
 		select {
 		case <-abort:
 			logger.Println("glob aborted")
 			return false
 		default:
 		}
-		if _, b := but[name]; !b {
-			vs = append(vs, name)
+
+		if _, ignore := but[pathInfo.Path]; ignore {
+			return true
+		}
+
+		if gp.TypeCb == nil { // no file type restrictions, common case
+			vs = append(vs, pathInfo.Path)
+		} else {
+			if (gp.TypeCb)(pathInfo.Info.Mode()) {
+				vs = append(vs, pathInfo.Path)
+			}
```

Comment:
This can be written a bit more succinctly as:

```go
if gp.TypeCb == nil || gp.TypeCb(pathInfo.Info.Mode()) {
    vs = append(vs, pathInfo.Path)
}
```